### PR TITLE
feat: Add `Averages` operation.

### PR DESCRIPTION
**Pull request by drupol**
This PR:

* [x] Add the `averages` operation
* [x] Is covered by unit tests
* [x] Has documentation

### CODE REVIEW

File: docs/pages/api.rst [hunk 1]
```diff
@@ -312,6 +312,18 @@ Signature: ``Collection::asyncMapN(callable ...$callbacks): Collection;``
     $collection = Collection::fromIterable(['c' => 3, 'b' => 2, 'a' => 1])
         ->asyncMapN($mapper1, $mapper2); // ['a' => 2, 'b' => 4, 'c' => 6]
 
+averages
+~~~~~~~~
+
+Calculate the average of a list of numbers.
```
Comment (AlexandruGG) [inline docs/pages/api.rst:318]: could you add a few more details to clarify what this means exactly?

I eventually understood after staring at the code and the example for a bit, but it's not immediately clear 😅.

Maybe we could mention that it's using `scanLeft1` under the hood and produces a list in the same fashion.

File: docs/pages/code/operations/averages.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+declare(strict_types=1);
+
+namespace App;
+
+use loophp\collection\Collection;
+
+include __DIR__ . '/../../../../vendor/autoload.php';
+
+$collection = Collection::fromIterable([1, 2, 3, 4, 5])
+    ->averages(); // [1, 1.5, 2, 2.5, 3]
```
Comment (AlexandruGG) [inline docs/pages/code/operations/averages.php:17]: here we could give an example of the calculations, like:

```php
// [1, 1.5, 2, 2.5, 3]
// [1/1, (1+2)/2, (1+2+3)/3 ...]
```

File: src/Contract/Operation/Averagesable.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+use loophp\collection\Contract\Collection;
+
+/**
+ * @template TKey
+ * @template T
+ */
+interface Averagesable
+{
+    /**
+     * Compute the averages.
```
Comment (AlexandruGG) [inline src/Contract/Operation/Averagesable.php:21]: the doc block here should match the description in the `api.rst`

File: src/Operation/Averages.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        return (new Pipe())()(
+            (new Normalize())(),
+            (new ScanLeft1())()(
+                /**
+                 * @param float $acc
+                 * @param float $value
+                 *
+                 * @return float
+                 */
+                static fn ($acc, $value, int $key) => ($acc * $key + $value) / ($key + 1)
```
Comment (AlexandruGG) [inline src/Operation/Averages.php:39]:
```suggestion
                static fn (float $acc, float $value, int $key): float => ($acc * $key + $value) / ($key + 1)
```

why not like this here? 🤔

Comment (drupol) [reply src/Operation/Averages.php]: Thanks: )

File: src/Contract/Operation/Averagesable.php (hunk 3 above)
Comment (drupol) [reply src/Contract/Operation/Averagesable.php]: Done

File: docs/pages/code/operations/averages.php (hunk 2 above)
Comment (drupol) [reply docs/pages/code/operations/averages.php]: Done.

File: docs/pages/api.rst (hunk 1 above)
Comment (drupol) [reply docs/pages/api.rst]: Done!

**Review by AlexandruGG — APPROVED** [review]
Really nice one 🚀

**Pull request merged**